e2e/test/scenarios/dashboard/reproductions/12926-editing-dashboards-doesnt-cancel-running-queries.cy.spec.js

```diff
@@ -0,0 +1,80 @@
+import {
+  editDashboard,
+  restore,
+  showDashboardCardActions,
+  undo,
+} from "e2e/support/helpers";
+
+const questionDetails = {
+  name: "Q1",
+  native: { query: "SELECT 1" },
+};
+
+describe("issue 12926", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  describe("card removal while query is in progress", () => {
+    it("should stop the ongoing query when removing a card from a dashboard", () => {
+      slowDownQuery();
+
+      cy.createNativeQuestionAndDashboard({
+        questionDetails,
+      }).then(({ body: { dashboard_id } }) => {
+        cy.visit(`/dashboard/${dashboard_id}`);
+      });
+
+      cy.window().then(win => {
+        cy.stub(win.XMLHttpRequest.prototype, "abort").as("xhrAbort");
+      });
+
+      removeCard();
+
+      cy.get("@xhrAbort").should("have.been.calledOnce");
+    });
+
+    it("should re-fetch the query when doing undo on the removal", () => {
+      slowDownQuery();
+
+      cy.createNativeQuestionAndDashboard({
+        questionDetails,
+      }).then(({ body: { dashboard_id } }) => {
+        cy.visit(`/dashboard/${dashboard_id}`);
+      });
+
+      removeCard();
+
+      restoreQuery();
+
+      undo();
+
+      cy.wait("@cardQueryRestored");
+    });
+  });
+});
+
+function slowDownQuery() {
+  cy.intercept("POST", "api/dashboard/*/dashcard/*/card/*/query", req => {
+    req.on("response", res => {
+      res.setDelay(10000);
+    });
+  }).as("cardQuerySlowed");
+}
+
+function restoreQuery() {
+  cy.intercept("POST", "api/dashboard/*/dashcard/*/card/*/query", req => {
+    req.continue();
+  }).as("cardQueryRestored");
+}
+
+function removeCard() {
+  editDashboard();
+
+  showDashboardCardActions();
+
+  cy.findByTestId("dashboardcard-actions-panel").within(() => {
+    cy.findByLabelText("close icon").click();
+  });
```

Comment:
nit: if there's only 1 action under `within()` you could chain it and that's preferable.

```suggestion
  cy.findByTestId("dashboardcard-actions-panel").findByLabelText("close icon").click();
```